src/requirements/requirement-exam-mapping.ts

```diff
@@ -0,0 +1,122 @@
+import examData, {
+  DEFAULT_COLLEGES,
+  ExamFulfillment,
+  ExamFulfillments,
+} from './data/exams/ExamCredit';
+import { colleges, College } from './data';
+
+type ExamRequirementsColleges = Record<number, College[]>;
+type ExamRequirements = {
+  colleges?: ExamRequirementsColleges; // if the user IS NOT in one of these colleges, the course id cannot fulfill the requirement
+  majorsExcluded?: string[]; // if the user IS in one of these majors, the course id cannot fulfill the requirement
+};
+
+/**
+ * Flat list of exam fulfillments, i.e.
+ * [
+ *   { courseId: 100, minimumScore: 4, credits: 4 },
+ *   { courseId: 101, minimumScore: 5, credits: 8 },
+ *   {
+ *     courseId: 102,
+ *     courseEquivalents: { DEFAULT: [Array], EN: [Array] },
+ *     minimumScore: 5,
+ *     credits: 4
+ *   },
+ *   ...
+ * ]
+ */
+const examFulfillmentList: ExamFulfillment[] = Object.values(examData)
+  .map((examFulfillments: ExamFulfillments) => Object.values(examFulfillments))
+  .flat(2);
+
+/**
+ * Mapping from exam id to requirements info (course equivalents and exam conditions), i.e.
```

Comment:
Nit - change "course equivalents and exam conditions" to "course equivalents and college and major conditions for which the exam fulfills" (or something along those lines)

Reply:
I get your point. I guess we can just call it `courseId` everywhere because there is no ambiguity for what a `courseId` means when associated with an exam.

> So I'm assuming you named this courseId in examData because it's the courseId equivalent for AP/IB? But they all are unique ids for the ap/ib exam it represents?

So yes to the first question, and almost to the second question. It's a unique id for each exam fulfillment (each exam can have more than one fulfillment if there are multiple scores it accepts). This is necessary because we want to use it as a representation of the exam in the requirements data, and different fulfillments (i.e. user scores) can fulfill requirements differently. The unique id for each exam is simply the exam name.